airbyte-config/init/src/main/java/io/airbyte/config/init/RemoteDefinitionsProvider.java

```diff
@@ -50,10 +50,10 @@ public void initialize() throws InterruptedException, IOException {
     final CombinedConnectorCatalog catalog = getRemoteDefinitionCatalog(this.remoteDefinitionCatalogUrl, this.timeout);
     this.sourceDefinitions = catalog.getSources().stream().collect(Collectors.toMap(
         StandardSourceDefinition::getSourceDefinitionId,
-        source -> source));
+        source -> source.withTombstone(source.getTombstone() != null && source.getTombstone())));
```

Comment:
I find this logic difficult to understand that it falls back to `false` when tombstone is `null`. 

nit: What do you think about keeping a bunch of definition static patching methods in a helper class + defining a `patchDefinition` method in this class that will call these helper functions? 
I would separate concerns and keep definition patching outside of initialize: 

```java
    this.sourceDefinitions = catalog.getSources().stream().collect(Collectors.toMap(
        StandardSourceDefinition::getSourceDefinitionId,
        source -> patchDefinition(source)));
    this.destinationDefinitions = catalog.getDestinations().stream().collect(Collectors.toMap(
        StandardDestinationDefinition::getDestinationDefinitionId,
        source -> patchDefinition(source)));
```

Reply:
👍🏻 moved to the existing JsonDefinitionsHelper class